components/elm/src/biogeochem/PhosphorusDynamicsMod.F90

```diff
@@ -648,122 +658,187 @@ subroutine PhosphorusBiochemMin_balance(bounds,num_soilc, filter_soilc, &
 
     ! set initial values for potential C and N fluxes
     biochem_pmin_ppools_vr_col(bounds%begc : bounds%endc, :, :) = 0._r8
-    biochem_pmin_to_plant_vr_patch(bounds%begp:bounds%endp,1:nlevdecomp) = 0._r8
-    biochem_pmin_to_plant_patch(bounds%begp:bounds%endp) = 0._r8
-      
-    do j = 1,nlevdecomp
-        do fc = 1,num_soilc
-            c = filter_soilc(fc)
+    
+    if(use_fates) then
+        ci = bounds%clump_index
+        max_comps = size(alm_fates%fates(ci)%bc_out(1)%cp_scalar,dim=1)
+        allocate(biochem_pmin_to_plant_vr_patch(max_comps,nlevdecomp))
+    else
+       allocate(biochem_pmin_to_plant_vr_patch(bounds%begp:bounds%endp,1:nlevdecomp))
+       biochem_pmin_to_plant_patch(:) = 0._r8
+    end if
+    biochem_pmin_to_plant_vr_patch(:,:) = 0._r8
+
+    do fc = 1,num_soilc
+        c = filter_soilc(fc)
+
+        if(use_fates) s = alm_fates%f2hmap(ci)%hsites(c)
+
+        biochem_pmin_to_plant(c) = 0._r8
+
+        do j = 1,nlevdecomp
+            
             biochem_pmin_vr(c,j) = 0.0_r8
             biochem_pmin_to_ecosysp_vr_col_pot(c,j) = 0._r8
-            do p = col_pp%pfti(c), col_pp%pftf(c)
-                if (veg_pp%active(p).and. (veg_pp%itype(p) .ne. noveg)) then
-                    !lamda_up = npimbalance(p) ! partial_vcmax/partial_lpc / partial_vcmax/partial_lnc
-                    lamda_up = cp_scalar(p)/max(cn_scalar(p),1e-20_r8)
+
+            if(use_fates) then
```

Comment:
`if (use_fates)` can be moved out of the do-loop

Reply:
good point, will do

Reply:
changed